overlord/devicestate/handlers_install.go

```diff
@@ -1321,6 +1321,12 @@ func (m *DeviceManager) doInstallFinish(t *state.Task, _ *tomb.Tomb) error {
 		Gadget:     snapInfos[snap.TypeGadget],
 		GadgetPath: snapSeeds[snap.TypeGadget].Path,
 
+		// modeenv:recovery_system will be set based on
+		// RecoverySystemDir and seeding reads this var
+		// from modeenv to know what it's seeding from
+		RecoverySystemDir:   systemLabel,
+		RecoverySystemLabel: systemLabel,
```

Comment:
the code is not  very consistent about these, we probably want to make it work to just set RecoverySystemLabel

Reply:
tweaked things around this